hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -1282,4 +1285,19 @@ public String createSnapshot(String pathStr, String snapshotName)
             ofsPath.getBucketName(),
             snapshotName);
   }
+
+  @Override
+  public SnapshotDiffReport getSnapshotDiffReport(Path snapshotDir,
+      String fromSnapshot, String toSnapshot)
+      throws IOException, InterruptedException {
+    OFSPath ofsPath = new OFSPath(snapshotDir, config);
+    SnapshotDiffResponse snapshotDiffResponse = null;
+    do {
+      snapshotDiffResponse = objectStore.snapshotDiff(ofsPath.getVolumeName(),
+          ofsPath.getBucketName(), fromSnapshot, toSnapshot, "", -1, false);
+      Thread.sleep(snapshotDiffResponse.getWaitTimeInMs());
+    } while (snapshotDiffResponse.getJobStatus() != DONE);
```

Comment:
Currently Distcp behaviour doesn't expect async behaviour for Snapshot Diff as we have currently implemented for Ozone i.e  it needs an instantaneous response with SnapshotDiffReport. Currently I have added a wait condition here in FS client of Ozone for it to work, 
Should this be okay or we should add the ability for distcp client to wait (which would require hdfs change similar to the first revision of https://github.com/apache/hadoop/pull/5364) ?

cc @prashantpogde @umamaheswararao

Reply:
I guess waiting here should be enough. The response returned is paginated, so we should be able to handle all the pages returned by the reponse.